.github/workflows/release.yml

```diff
@@ -1,16 +1,37 @@
-name: release
+name: 'New Release'
+run-name: 'Release ${{ inputs.version_number }}'
+
 on:
-  push:
-    tags:
-      - 'v*'
-jobs: 
-  goreleaser:
+  workflow_dispatch:
+    inputs:
+      version_number:
+        description: 'Version number (e.g., v1.0.0, v1.0.0-pre, v1.0.0-pre1)'
+        required: true
+      skip_tests:
+        description: 'Skip QA acceptance tests, define value to `true` to explicitly skip'
+
+jobs:
+    # QA acceptance tests are skipped for pre releases or when explicit input parameter is used
+  run-qa-acceptance-tests:
+    if: inputs.skip_tests != 'true' && !contains(inputs.version_number, 'pre')
+    secrets: inherit
+    uses: ./.github/workflows/acceptance-tests.yml
+    with:
+      atlas_cloud_env: "qa"
+
+  release:
     runs-on: ubuntu-latest
+    needs: [ run-qa-acceptance-tests ]
+    if: ${{ always() && (needs.run-qa-acceptance-tests.result == 'skipped' || needs.run-qa-acceptance-tests.result == 'success') }}
     steps:
       - name: Checkout
         uses: actions/checkout@v4
       - name: Unshallow
         run: git fetch --prune --unshallow
+      - name: Create release tag
```

Comment:
Do we want to add the logic to delete the tag in case of failure of a step after the tag creation?

Reply:
This can be an improvement but not sure if worth including in this PR. This case would only happen if `ghaction-import-gpg` or `goreleaser-action` action fails, which does not seem frequent from previous releases.

Reply:
I'd suggest documenting what we should do in case of a failure during the go releaser step in this PR: manually delete the tag and try again to run the action. Feel free to create a ticket to add this improvement if you don't want to do it as part of this PR.

Reply:
leaving created jira ticket just for reference: https://jira.mongodb.org/browse/CLOUDP-221414